### DISCUSSION TITLE
issue: unable to use type safe paths without a cast when using a generic template argument to useForm

### DISCUSSION THREAD

**Discussion by jacquesg** [react-hook-form/react-hook-form#10614] [Q&A]
### Version Number

7.43.9

### Codesandbox/Expo snack

https://codesandbox.io/s/elegant-joji-ldcs94?file=/src/App.tsx

### Steps to reproduce

Pass a generic argument that extends another base type as the argument to `useForm`.

### Expected behaviour

No typescript error expect on the following line:
`<input {...register("firstName")} placeholder="First Name" />`

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Relevant log output

```shell
Argument of type '"firstName"' is not assignable to parameter of type 'Path<TFieldValues>'.ts(2345)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jacquesg** [comment]
The only way I know of to make this error go away is to cast `"firstName"` to `"firstName" as Path<TFieldValues>`. This kind of defeats the purpose of having type-safety here, as it will just silence a typo for example in the field name.

**Comment by bluebill1049** [comment]
what you provided for example is currently not possible for the component to infer the type.  I don't think it's even possible to infer the type a generic with a template string literal from typescript. If you find a way to achieve that, let us know or send us a PR.

**Reply by jacquesg** [reply]
I get that it isn't possible for all of the keys inside the generic type, makes perfect sense that any keys that are not present in the base type. The keys in the base type should be known though?

**Comment by jacquesg** [comment]
If this isn't possible, are there any recommended patterns here to change the code to not lose type safety?
